app/models/firm.rb

```diff
@@ -151,6 +165,11 @@ def advice_types
     ADVICE_TYPES_ATTRIBUTES.map { |a| [a, self[a]] }.to_h
   end
 
+  def primary_advice_method
```

Comment:
It's used by RAD too — see here for instance: https://github.com/moneyadviceservice/rad/pull/275/files#diff-6004437aeea98d71ab51826159b8414fR8

Nice idea though — not used custom validators before! Will be sure to in future as big messy validations are one of my least favourite things.